work_queue/src/work_queue_process.c

```diff
@@ -268,4 +269,73 @@ void work_queue_process_kill(struct work_queue_process *p)
 	waitpid(p->pid, NULL, 0);
 }
 
+
+static int download_url(const char *url, const char *cache_name) {
+	debug(D_WQ, "Retrieving %s from (%s)\n", cache_name, url);
+
+	char command[WORK_QUEUE_LINE_MAX];
+	snprintf(command, WORK_QUEUE_LINE_MAX, "curl -f -o \"cache/%s\" \"%s\"", cache_name, url);
+
```

Comment:
I predict that sooner or later, we wlil run on a machine where _curl_ is unavailable, and mysterious errors will result.  Suggest that you _debug()_ the command line and check for curl in the path.